src/main.rs

```diff
@@ -36,8 +36,25 @@ fn mount_parse_ovf_env(dev: String) -> Result<Environment, anyhow::Error> {
     Ok(environment)
 }
 
+fn get_environment() -> Result<Environment, anyhow::Error> {
```

Comment:
I think this refactor is a fine improvement as-is.

However, it might make sense to move this functionality to the library as a convenience function. `mount_parse_ovf_env` is itself a bit of a convenience function for the media module. I think it would make sense to move `media::parse_ovf_env` to be in `impl Media<Mounted>`, probably combining it with `read_ovf_env_to_string` unless there's a good case someone wants the string rather than the parsed `Environment` object. The combined function could probably just be called `environment` or something like that.

Whether or not you want to take that on in this PR is up to you since I think this is clearer either way.

Reply:
Sounds like a good idea.
However when `parse_ovf_env` is in `impl Media<Mounted>`, then unit tests like `test_get_ovf_env_none_missing` would not be able to find the function.
Let me think.

Reply:
I was not able to move `parse_ovf_env` into `impl Media<Mounted>`, due to the unit tests as mentioned above. That's not impossible to do in theory, like creating mount before testing in each unit test, etc., but even after that I hit other issues of permission denied, so seems that it is not worth doing so.

Instead, as you suggested, I moved `mount_parse_ovf_env` to libazureinit/media.